static/js/lib/api.js

```diff
@@ -55,13 +50,12 @@ export function patchUserProfile(
   })
 }
 
-export async function getDashboard(username: string): Promise<Dashboard> {
+export async function getDashboard(username: string): Promise<Response> {
   try {
     const response = await fetchJSONWithCSRF(`/api/v0/dashboard/${username}/`)
     return response
   } catch (response) {
-    loginOnEdXError(response)
-    return response
+    return loginOnEdXError(response)
```

Comment:
I think with this function (and the `getPrograms` function below) it might be better to return `Promise.reject(response)` directly, so that we will see a Sentry error which has information about what went wrong (instead of just the generic 'Unable to process'). What do you think?